docs: Parameters example

amirkaws opened the pull request:
Issue number: #286 

## Summary

### Changes

Introduce an example for Parameters utility.

### User experience

This changes added an example solution on how to use Parameters utility

## Checklist

Please leave checklist items unchecked if they do not apply to your change.

* [x] [Meets tenets criteria](https://awslabs.github.io/aws-lambda-powertools-dotnet/tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-dotnet/blob/develop/.github/semantic.yml)


<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

sliedig commented on examples/Parameters/README.md:
Please add instructions on how to deploy dependencies

amirkaws replied:
done

sliedig commented on examples/Parameters/template.yaml:
```diff
@@ -0,0 +1,294 @@
+      MultiParamTableName: PowertoolsParametersMultiple
+      MultiParamId: my-hash-key
+    SSM:
+      SingleParamName: /powertools/path/prefix/parameter
```
rename to /powertools-examples/path/prefix/parameter

sliedig commented on examples/Parameters/template.yaml:
```diff
@@ -0,0 +1,294 @@
+      MultiParamId: my-hash-key
+    SSM:
+      SingleParamName: /powertools/path/prefix/parameter
+      OtherParamName: /powertools/path/prefix/other-parameter
```
/powertools-examples/path/prefix/other-parameter

sliedig commented on examples/Parameters/template.yaml:
```diff
@@ -0,0 +1,294 @@
+    SSM:
+      SingleParamName: /powertools/path/prefix/parameter
+      OtherParamName: /powertools/path/prefix/other-parameter
+      MultiParamPrefix: /powertools/path/prefix
```
/powertools-examples/path/prefix

amirkaws replied:
renamed it to /powertools-parameters-example/path/prefix

sliedig commented on examples/Parameters/template.yaml:
```diff
@@ -0,0 +1,294 @@
+Mappings:
+  Constants:
+    DynamoDB:
+      SingleParamTableName: PowertoolsParametersSingle
```
rename to PowertoolsParametersExampleSingle

sliedig commented on examples/Parameters/template.yaml:
```diff
@@ -0,0 +1,294 @@
+    DynamoDB:
+      SingleParamTableName: PowertoolsParametersSingle
+      SingleParamId: my-parameter
+      MultiParamTableName: PowertoolsParametersMultiple
```
rename to PowertoolsParametersExampleMultiple

amirkaws replied:
done

sliedig reviewed (changes requested):
@amirkaws just a couple of comments, nothing major. Just need to provide a bit more context for the demo

amirkaws commented:
> @amirkaws just a couple of comments, nothing major. Just need to provide a bit more context for the demo

@sliedig fyi, I have added a section in readme for the additional resources are created and where to find them, also addressed other comments.
